frontend/src/metabase/embedding/embedding-iframe-sdk-setup/utils/build-embed-attributes.unit.spec.ts

```diff
@@ -0,0 +1,539 @@
+import type {
+  SdkIframeEmbedSetupExperience,
+  SdkIframeEmbedSetupSettings,
+} from "metabase/embedding/embedding-iframe-sdk-setup/types";
+
+import {
+  buildEmbedAttributes,
+  formatAttributeValue,
+} from "./build-embed-attributes";
+
+// Mock getVisibleParameters
+jest.mock("./get-visible-parameters", () => ({
+  getVisibleParameters: jest.fn((params, locked) => {
+    if (!params || !locked?.length) {
+      return params;
+    }
+    const filtered = Object.fromEntries(
+      Object.entries(params).filter(([key]) => !locked.includes(key)),
+    );
+    return Object.keys(filtered).length > 0 ? filtered : undefined;
+  }),
+}));
+
+describe("buildEmbedAttributes", () => {
```

Comment:
I think the test is a bit overboard. If we rely on `ALLOWED_EMBED_SETTING_KEYS_MAP` to build attributes, then we just need one or two test to see if values outside `ALLOWED_EMBED_SETTING_KEYS_MAP[experience]` is not allowed, and values within it is being dash-cased properly, and maybe one for guest embed's parameters (special case).